hal/shared/demux.cpp

```diff
@@ -38,17 +41,33 @@ int Demux::write(uint8_t pin, uint8_t value) {
     DemuxLock lock();
     CHECK_TRUE(pin < DEMUX_MAX_PIN_COUNT && pin != 0, SYSTEM_ERROR_INVALID_ARGUMENT); // Y0 is not available for user's usage.
     CHECK_TRUE(initialized_, SYSTEM_ERROR_INVALID_STATE);
+#if HAL_PLATFORM_NRF52840
     // We can do it in this way for now, since the control pins are continuous.
     uint32_t currOut = (nrf_gpio_port_out_read(DEMUX_NRF_PORT) >> DEMUX_PINS_SHIFT) & 0x00000007;
     if ((currOut == pin && value == 0) || (currOut != pin && value == 1)) {
         return SYSTEM_ERROR_NONE;
     }
+#endif
+
     if (value) {
+#if HAL_PLATFORM_NRF52840
         // Select Y0 by default, so that all other pins output high.
         nrf_gpio_port_out_clear(DEMUX_NRF_PORT, DEMUX_PIN_2_MASK | DEMUX_PIN_1_MASK | DEMUX_PIN_0_MASK);
+#elif HAL_PLATFORM_RTL872X
+        // Select Y0 by default, so that all other pins output high.
+        hal_gpio_write(DEMUX_C, 0);
+        hal_gpio_write(DEMUX_B, 0);
+        hal_gpio_write(DEMUX_A, 0);
+#endif
         setPinValue(0, 0);
     } else {
+#if HAL_PLATFORM_NRF52840
         nrf_gpio_port_out_set(DEMUX_NRF_PORT, ((uint32_t)pin) << DEMUX_PINS_SHIFT);
+#elif HAL_PLATFORM_RTL872X
+        hal_gpio_write(DEMUX_C, pin >> 2);
```

Comment:
The 2nd revision to eval board puts the required pins on the same port, so I'd prefer to defer changes here until then.

Reply:
Since this will get merged to develop, I would think we should address them here as well, and change them as needed later. I added
```
        hal_gpio_write(DEMUX_C, (pin & DEMUX_PIN_2_MASK) ? 1 : 0);
        hal_gpio_write(DEMUX_B, (pin & DEMUX_PIN_1_MASK) ? 1 : 0);
        hal_gpio_write(DEMUX_A, (pin & DEMUX_PIN_0_MASK) ? 1 : 0);
```

Reply:
This works too, but I didn't see these defined, so something like this needs to go somewhere:
```
#define DEMUX_PIN_2_MASK (1 << 2)
#define DEMUX_PIN_1_MASK (1 << 1)
#define DEMUX_PIN_0_MASK (1 << 0)
```